pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -211,9 +211,14 @@ func (a Adapter) Build(parameters common.BuildParameters) (err error) {
 	}
 
 	if !isImageRegistryInternal {
-		if err := a.createDockerConfigSecret(parameters); err != nil {
+		dockerConfigSecretBytes, err := a.createDockerConfigSecretBytes(parameters.DockerConfigJSONFilename)
+		if err != nil {
 			return err
 		}
+
+		if _, err := a.createDockerConfigSecret(regcredName, parameters.EnvSpecificInfo.GetNamespace(), dockerConfigSecretBytes); err != nil {
+			return errors.Wrap(err, "failed to create dockerconfig secret")
+		}
 	}
```

Comment:
The original code is definitely cleaner.   I would leave it as a.createDockerConfigSecret()

Reply:
would it be better if I renamed ```createDockerConfigSecretBytes()``` to something else, and added a call to existing ```createDockerConfigSecret()``` in that function at the end? (Same in kaniko_build.go)
That way we can have it be called something similar to ```createDockerConfigSecret``` while reusing the create secret logic

Reply:
It is not a function naming issue.   The caller does not care about the byte array of the secret.   The caller only cares about getting secrete created on the cluster.    The unit test should be validating that a.createDockerConfigSecret() has successfully created an expected secret if I provide a docker config file path in the parameters.

Reply:
reverted to old version